core/server/common/src/main/java/alluxio/master/journal/AsyncJournalWriter.java

```diff
@@ -308,6 +316,9 @@ private void doFlush() {
           }
         }
       } catch (IOException | JournalClosedException exc) {
+        // Add the error logging here since the actual flush error may be overwritten
+        // by the future meaningless ratis.protocol.AlreadyClosedException
+        SAMPLING_LOG.warn("Failed to flush journal entry", exc);
```

Comment:
NIT: the sampling logger works by treating every log template as a separate error. I don't know if this will be useful, but if we use the template `SAMPLING_LOG.warn("Failed to flush journal entry: " + exc.getMessage(), exc); `, we would get to see the warning log of each cause, once every 30s. This may help if there are potentially different causes of journal flush errors, and we would still not miss any cause.

Reply:
Good catch! in the previous way, the different error messages may still be ignored.